src/charm.py

```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env python3
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+#
+# Learn more at: https://juju.is/docs/sdk
+
+"""Integrator charm to provide a fiveg_gnb_identity."""
+
+import logging
+from typing import Optional
+
+import ops
+from charms.sdcore_gnbsim_k8s.v0.fiveg_gnb_identity import (  # type: ignore[import]
+    GnbIdentityProvides,
+)
+from ops.model import ActiveStatus, BlockedStatus
+
+# Log messages can be retrieved using juju debug-log
+
+logger = logging.getLogger(__name__)
+
+GNB_IDENTITY_RELATION_NAME = "fiveg_gnb_identity"
+
+
+class SdcoreGnbIntegratorCharm(ops.CharmBase):
+    """Charm for gNB Integrator Service."""
+
+    def __init__(self, *args):
+        super().__init__(*args)
+        self.framework.observe(self.on.config_changed, self._on_config_changed)
+        self._gnb_identity_provider = GnbIdentityProvides(self, GNB_IDENTITY_RELATION_NAME)
+        self.framework.observe(
+            self._gnb_identity_provider.on.fiveg_gnb_identity_request,
+            self._on_config_changed,
+        )
+
+    def _on_config_changed(self, event: ops.ConfigChangedEvent):
+        """Handle changed configuration.
+
+        Verifies the configuration is valid, and if so, notifies
+        all integrations of the current configuration.
+
+        Learn more about config at https://juju.is/docs/sdk/config
+        """
+        if invalid_configs := self._get_invalid_configs():
+            self.unit.status = BlockedStatus(f"Configurations are invalid: {invalid_configs}")
+            return
+
+        self._update_fiveg_gnb_identity_relation_data()
+        self.unit.status = ActiveStatus()
```

Comment:
Let's start by using `collect-status` instead of ading it as a follow-up improvement

Reply:
done